tests/e2e/util.go

```diff
@@ -7029,3 +7025,268 @@ func removeStoragePolicyQuota(ctx context.Context, restClientConfig *rest.Config
+	cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restClientConfig, cnsoperatorv1alpha1.GroupName)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	spq := &storagepolicyv1alpha2.StoragePolicyQuota{}
+	err = cnsOperatorClient.Get(ctx,
+		pkgtypes.NamespacedName{Name: scName + storagePolicyQuota, Namespace: namespace}, spq)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	scLevelQuotaStatusList := spq.Status.SCLevelQuotaStatuses
+	for _, item := range scLevelQuotaStatusList {
+		if item.StorageClassName == scName {
+			usedQuota = item.SCLevelQuotaUsage.Used
+			reservedQuota = item.SCLevelQuotaUsage.Reserved
+			ginkgo.By(fmt.Sprintf("usedQuota %v, reservedQuota %v", usedQuota, reservedQuota))
+		}
+	}
+	return usedQuota, reservedQuota
+}
+
+// Get getStoragePolicyUsageForSpecificResourceType based on resourceType (i.e., either volume, snapshot, vmservice)
+// resourceUsage will be either pvcUsage, vmUsage and snapshotUsage
+func getStoragePolicyUsageForSpecificResourceType(ctx context.Context, restClientConfig *rest.Config,
+	scName string, namespace string, resourceUsage string) (*resource.Quantity, *resource.Quantity) {
+	var usedQuota, reservedQuota *resource.Quantity
+	cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restClientConfig, cnsoperatorv1alpha1.GroupName)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	//spq := &storagepolicyv1alpha2.StoragePolicyQuota{}
+	spq := &storagepolicyv1alpha2.StoragePolicyUsage{}
+	err = cnsOperatorClient.Get(ctx,
+		pkgtypes.NamespacedName{Name: scName + resourceUsage, Namespace: namespace}, spq)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	if spq.Status.ResourceTypeLevelQuotaUsage.DeepCopy() == nil {
+		zeroQuantity := resource.MustParse("0")
+		usedQuota = &zeroQuantity
+		reservedQuota = &zeroQuantity
+	} else {
+		usedQuota = spq.Status.ResourceTypeLevelQuotaUsage.Used
+		reservedQuota = spq.Status.ResourceTypeLevelQuotaUsage.Reserved
+	}
+	return usedQuota, reservedQuota
+}
+
+func validate_totalStoragequota(ctx context.Context, diskSize int64, totalUsedQuotaBefore *resource.Quantity,
+	totalUsedQuotaAfter *resource.Quantity) bool {
+	var validTotalQuota bool
+	validTotalQuota = false
+
+	out := make([]byte, 0, 64)
+	result, suffix := totalUsedQuotaBefore.CanonicalizeBytes(out)
+	value := string(result)
+	quotaBefore, err := strconv.ParseInt(string(value), 10, 64)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By(fmt.Sprintf(" quotaBefore :  %v%s", quotaBefore, string(suffix)))
+
+	result1, suffix1 := totalUsedQuotaAfter.CanonicalizeBytes(out)
+	value1 := string(result1)
+	quotaAfter, err := strconv.ParseInt(string(value1), 10, 64)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By(fmt.Sprintf(" quotaAfter :  %v%s", quotaAfter, string(suffix1)))
+
+	if string(suffix) != string(suffix1) {
+		if string(suffix1) == "Mi" {
+			var bytes int64 = quotaBefore
+			// Convert to Gi
+			//kibibytes := float64(bytes) / 1024
```

Comment:
remove all commented code if not required

Reply:
This is like info